Tests/MockSession.swift

```diff
@@ -35,16 +35,18 @@ open class MockSession: SessionPublisherProtocol {
   public func dataTaskPublisher(for urlRequest: URLRequest) -> AnyPublisher<(data: Data, response: URLResponse), Error> {
     methodCallStack.append(#function)
     finalUrlRequest = urlRequest
-    return Future<(data: Data, response: URLResponse), Error> { promise in
-      usleep(20)
-      if let successResponse = self.result?.0 {
-        promise(.success((successResponse.0,
-                          HTTPURLResponse(url: URL(string: "foo.com")!,
-                                          statusCode: successResponse.1,
-                                          httpVersion: nil,
-                                          headerFields: nil)!)))
-      } else if let errorResponse = self.result?.1 {
-        promise(.failure(NetworkError.urlError(errorResponse)))
+    return Deferred {
+      Future<(data: Data, response: URLResponse), Error> { promise in
+        usleep(20)
```

Comment:
why do we need this timeout?

Reply:
I put it just to simulate a slower execution in tests. It has nothing to do with the production code. I wanted to execute them slightly after each other. I guess it can be removed.